### PR TITLE
Add initial naive implementation of pitch detection

### PR DESCRIPTION
**Pull request by Holzhaus**

### CODE REVIEW

File: src/pitch.rs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        //
+        //  a
+        //   \
+        //  ─────
+        //     \
+        //      b
+        //
+        // We could now assume that the zero crossing happened just now, but it's more precise to
+        // interpolate the exact subsample position where zero was crossed. This can be done by
+        // assuming that the zero crossing position is equal to the proportion how far both are
```
Comment (Be-ing) [inline src/pitch.rs:45]: IIUC, this is assuming there is no acceleration? But isn't acceleration what we're trying to detect?

File: src/timecode.rs [hunk 2]
```diff
@@ -21,6 +24,7 @@ pub struct TimecodeChannel {
 }
 
 const SAMPLE_RATE_HZ: f64 = 44100.0;
+const TIMECODE_FREQUENCY_HZ: f64 = 1000.0;
```
Comment (Be-ing) [inline src/timecode.rs:27]: This should be a property of the TimecodeFormat.

File: src/pitch.rs (hunk 1 above)
Comment (Holzhaus) [reply src/pitch.rs:45]: Yes, but this is between 2 consecutive samples, i.e. between 1/44100 of a second. I think a linear interpolation suffices.

Comment (Be-ing) [reply src/pitch.rs:45]: It probably does, but nevertheless I think it would be good to document this.

**Pull request merged**